compiler/noirc_frontend/src/hir/comptime/interpreter.rs

```diff
@@ -1741,6 +1742,34 @@ impl<'local, 'interner> Interpreter<'local, 'interner> {
         Ok(Value::Unit)
     }
 
+    fn evaluate_loop(&mut self, expr: ExprId) -> IResult<Value> {
+        let was_in_loop = std::mem::replace(&mut self.in_loop, true);
+        let in_lsp = self.elaborator.interner.is_in_lsp_mode();
+        let mut counter = 0;
+
+        loop {
+            self.push_scope();
+
+            match self.evaluate(expr) {
+                Ok(_) => (),
+                Err(InterpreterError::Break) => break,
```

Comment:
Shouldn't `pop_scope` be called before we `break`? Otherwise we exit `evaluate_loop` and leave an extra entry in `comptime_scopes`.